src/components/carousel/daikin-carousel.ts

```diff
@@ -0,0 +1,274 @@
+ *
+ * @example
+ *
+ * ```html
+ * <daikin-carousel>
+ *   <daikin-carousel-item>Carousel item 1</daikin-carousel-item>
+ *   <daikin-carousel-item>Carousel item 2</daikin-carousel-item>
+ *   <daikin-carousel-item>Carousel item 3</daikin-carousel-item>
+ * </daikin-carousel>
+ * ```
+ */
+@customElement("daikin-carousel")
+export class DaikinCarousel extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: block;
+      width: 100%;
+      position: relative;
+
+      --ddc-transition-duration: ${unsafeCSS(DEFAULT_TRANSITION_DURATION)};
+    }
+  `;
+
+  /**
+   * The index number of the content currently being displayed.
+   */
+  @property({ type: Number, reflect: true, attribute: "current-index" })
+  currentIndex = 0;
+
+  /**
+   * Specify the variant of the control button.
+   */
+  @property({
+    type: String,
+    reflect: true,
+    attribute: "control-button-variant",
+  })
+  controlButtonVariant: "ghost" | "fill" = "ghost";
+
+  /**
+   * Whether or not to allow content to be moved by swiping.
+   */
+  @property({ type: Boolean, reflect: true, attribute: "allow-swipe" })
+  allowSwipe: boolean = false;
+
+  @queryAssignedElements({ selector: "daikin-carousel-item" })
+  private readonly _items!: readonly DaikinCarouselItem[];
+
+  @state()
+  private _itemCount: number = 0;
+
+  private _prevButton = createRef<HTMLButtonElement>();
+  private _nextButton = createRef<HTMLButtonElement>();
+  private _indicatorButtonsContainer = createRef<HTMLElement>();
+
+  private _emitSelect(
+    operation: "prev" | "next" | "indicator",
+    beforeCurrentIndex: number
+  ) {
+    this.dispatchEvent(
+      new CustomEvent("select", {
+        detail: { operation, beforeCurrentIndex },
+      })
+    );
+  }
+
+  private _handleClickIndicator(index: number) {
+    const beforeCurrentIndex = this.currentIndex;
+
+    this.currentIndex = index;
+    this._emitSelect("indicator", beforeCurrentIndex);
+  }
+
+  private _handleKeydownIndicator(event: KeyboardEvent): void {
+    const moveOffset = (
+      {
+        ArrowRight: 1,
+        ArrowLeft: -1,
+      } as const
+    )[event.key];
+
+    if (!moveOffset) {
+      return;
+    }
+
+    const buttons = [
+      ...(this._indicatorButtonsContainer.value?.querySelectorAll("button") ??
+        []),
+    ];
+
+    this._moveBy(moveOffset);
+    buttons[this.currentIndex].focus();
+  }
+
+  private _handleSlotchange() {
+    const items = this._items;
+
+    this._itemCount = items.length;
+    for (const [index, item] of items.entries()) {
+      item.label = `${index + 1} of ${this._itemCount}`;
+    }
+
+    this._updateItemActive();
+  }
+
+  private _moveBy(moveOffset: 1 | -1) {
+    const newIndex = this.currentIndex + moveOffset;
+    if (
+      (newIndex < 0 && moveOffset === -1) ||
+      (newIndex >= this._itemCount && moveOffset === 1)
+    ) {
```

Comment:
Now we don't have to look `moveOffset`.

```suggestion
    if (newIndex < 0 || newIndex >= this._itemCount) {
```